compiler/il/OMRBlock.cpp

```diff
@@ -1472,11 +1472,35 @@ OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyE
             // If we can not find a register, store it into temp slot.
             if (!iter->second.second)
                {
-               iter->second.second = TR::Node::createWithSymRef(value, comp->il.opCodeForDirectLoad(value->getDataType()), 0, ref);
+               TR::DataType valDataType = value->getDataType();
+               TR::Node *convertedValueNode  = value;
+               // In case for a given node, if the load/store (Minimum 4 bytes for alignment) size in memory is not same as node's data type,
+               // We need to convert the node to the type which can be stored into memory. Also we need to make sure that replacement node is a
+               // node that is converted back to original type of the node.
+               if (comp->fe()->dataTypeForLoadOrStore(valDataType) != valDataType)
+                  {
+                  TR::ILOpCodes convertOpCode = TR::ILOpCode::getProperConversion(valDataType, comp->fe()->dataTypeForLoadOrStore(valDataType), false);
```

Comment:
can the two data types here please be pulled into helpfully named locals and use those locals both here on 1482 and on 1495 - I worry that you could endup with some kind of weird asymmetry and using the local will ensure we keep everything straight and keep it easy to see what is gong on while also reducing the possibility of mistakes relying on nodes to report their type (shouldn't be needed, but I think the locals would definitely be more clear).

Reply:
Made changes in https://github.com/eclipse/omr/pull/4551/commits/5712d5e16bfc4c66ac19c900086f36b788d1731c